kvpress/presses/simlayerkv_press.py

```diff
@@ -0,0 +1,89 @@
+from dataclasses import dataclass
+
+import torch
+from torch import nn
+from transformers import QuantizedCache
+
+from kvpress.presses.base_press import BasePress
+from kvpress.presses.snapkv_press import SnapKVPress
+
+
+@dataclass
+class SimLayerKVPress(BasePress):
+    """
+    SimLayerKV (https://arxiv.org/abs/2410.13846) uses a layer-wise approach to compression:
+        - layers identified as lazy use the Streaming LLM approach (only initial and recent KV pairs are kept)
+        - other layers use the full KV cache
+
+    To identify lazy layers, the last attention weights are used. If the sum of attention weights of the last tokens
+    over the initial and recent tokens is above the lazy_threshold, the layer is considered lazy.
+
+    Official implementation: https://github.com/sail-sg/SimLayerKV. We use n_last=1 to match SKLV-decode
+    """
+
+    lazy_threshold: float
+    n_last: int = 1
+    n_recent: int = 1024
+    n_initial: int = 4
+
+    def is_lazy(
+        self,
+        module: nn.Module,
+        hidden_states: torch.Tensor,
+        keys: torch.Tensor,
+    ) -> bool:
+        """
+        Compute the average attention weights of the last tokens over the initial and recent tokens.
+        A slight difference with the original implementation is that we
```

Comment:
Docstring isn't finished here.
Could you also add an explanation what "is_lazy" refers to.

Reply:
better ?